libdnf/module/nsvcap.cpp

```diff
@@ -0,0 +1,140 @@
+    std::regex("^" MODULE_NAME ":" MODULE_STREAM     "()"               "()"           "::"  MODULE_ARCH "\\/?" "()"           "$", std::regex::extended),
+    std::regex("^" MODULE_NAME ":" MODULE_STREAM ":" MODULE_VERSION ":" MODULE_CONTEXT       "()"        "\\/"  MODULE_PROFILE "$", std::regex::extended),
+    std::regex("^" MODULE_NAME ":" MODULE_STREAM ":" MODULE_VERSION     "()"                 "()"        "\\/"  MODULE_PROFILE "$", std::regex::extended),
+    std::regex("^" MODULE_NAME ":" MODULE_STREAM ":" MODULE_VERSION ":" MODULE_CONTEXT       "()"        "\\/?" "()"           "$", std::regex::extended),
+    std::regex("^" MODULE_NAME ":" MODULE_STREAM ":" MODULE_VERSION     "()"                 "()"        "\\/?" "()"           "$", std::regex::extended),
+    std::regex("^" MODULE_NAME ":" MODULE_STREAM     "()"               "()"                 "()"        "\\/"  MODULE_PROFILE "$", std::regex::extended),
+    std::regex("^" MODULE_NAME ":" MODULE_STREAM     "()"               "()"                 "()"        "\\/?" "()"           "$", std::regex::extended),
+    std::regex("^" MODULE_NAME     "()"              "()"               "()"           "::"  MODULE_ARCH "\\/"  MODULE_PROFILE "$", std::regex::extended),
+    std::regex("^" MODULE_NAME     "()"              "()"               "()"           "::"  MODULE_ARCH "\\/?" "()"           "$", std::regex::extended),
+    std::regex("^" MODULE_NAME     "()"              "()"               "()"                 "()"        "\\/"  MODULE_PROFILE "$", std::regex::extended),
+    std::regex("^" MODULE_NAME     "()"              "()"               "()"                 "()"        "\\/?" "()"           "$", std::regex::extended),
+    std::regex("^" MODULE_NAME ":" MODULE_STREAM     "()"           ":" MODULE_CONTEXT       "()"        "\\/?" MODULE_PROFILE "$", std::regex::extended),
+    std::regex("^" MODULE_NAME ":" MODULE_STREAM     "()"           ":" MODULE_CONTEXT       "()"        "\\/?" "()"           "$", std::regex::extended)};
+// clang-format on
+
+
+static const std::vector<Nsvcap::Form> MODULE_SPEC_FORMS{
+    Nsvcap::Form::N,
+    Nsvcap::Form::NP,
+    Nsvcap::Form::NS,
+    Nsvcap::Form::NSP,
+    Nsvcap::Form::NSC,
+    Nsvcap::Form::NSCP,
+    Nsvcap::Form::NSV,
+    Nsvcap::Form::NSVP,
+    Nsvcap::Form::NSVC,
+    Nsvcap::Form::NSVCP,
+    Nsvcap::Form::NA,
+    Nsvcap::Form::NAP,
+    Nsvcap::Form::NSA,
+    Nsvcap::Form::NSAP,
+    Nsvcap::Form::NSVA,
+    Nsvcap::Form::NSVAP,
+    Nsvcap::Form::NSVCA,
+    Nsvcap::Form::NSVCAP};
+
+
+const std::vector<Nsvcap::Form> & Nsvcap::get_default_module_spec_forms() {
+    return MODULE_SPEC_FORMS;
+}
+
+
+bool Nsvcap::parse(const std::string nsvcap_str, Nsvcap::Form form) {
+    enum { NAME = 1, STREAM = 2, VERSION = 3, CONTEXT = 4, ARCH = 5, PROFILE = 6, _LAST_ };
+    std::smatch matched_result;
+    auto matched = std::regex_match(nsvcap_str, matched_result, NSVCAP_FORM_REGEX[static_cast<unsigned>(form) - 1]);
```

Comment:
I worry this std::regex_match() call will suffer from the same issue as I fixed for RPM in #268.

Reply:
Correct, but creating an issue is sufficient.

Reply:
I think we can resolve it the same way: https://github.com/rpm-software-management/dnf5/pull/299